### PR TITLE
Added responsiveness to dashboard, header and footer

### LINKED ISSUE #155: Make adaptive design for `dashboard` page
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Make adaptive design for `phone` and `tablet` for `dashboard` page
3. This task only for project contributors not for new comers

### PR DESCRIPTION
**Pull request by PoulavBhowmick03**
Fixes #155 

https://github.com/user-attachments/assets/4d2ec3cd-f561-454f-a73d-d1afb18bc134

### CODE REVIEW

File: frontend/src/components/header/Header.jsx [hunk 1]
```diff
@@ -1,10 +1,83 @@
...
+    } else {
+      document.body.style.overflow = 'unset';
+    }
+    return () => {
+      document.body.style.overflow = 'unset';
+    };
+  }, [isMenuOpen]);
+
+  const toggleMenu = () => {
+    setIsMenuOpen(!isMenuOpen);
```
Comment (whateverfw) [inline frontend/src/components/header/Header.jsx:28]: could you please create custom hook in `hooks` directory called `useLockBodyScroll` and basically use the exact hook, that was recently created (not merged yet) https://github.com/djeck1432/spotnet/pull/167/files#diff-e3912185f39bc0891d631dd214c08db009ca6edbcbd12a1717fa1afec9b82769
Thx

File: frontend/src/components/header/Header.jsx [hunk 2]
```diff
@@ -1,10 +1,83 @@
...
+      </NavLink>
+      <NavLink 
+        to="/dashboard" 
+        className={({ isActive }) => (isActive ? 'active-link' : '')}
+        onClick={() => setIsMenuOpen(false)}
+      >
+        Dashboard
+      </NavLink>
+    </div>
+  );
```
Comment (whateverfw) [inline frontend/src/components/header/Header.jsx:49]: This component should be in a separate file

File: frontend/src/components/header/Header.jsx [hunk 3]
```diff
@@ -1,10 +1,83 @@
...
+            onConnectWallet();
+            setIsMenuOpen(false);
+          }}
+        >
+          <span>Connect Wallet</span>
+        </button>
+      )}
+    </div>
+  );
+
```
Comment (whateverfw) [inline frontend/src/components/header/Header.jsx:80]: This component should be in a separate file

File: frontend/src/components/header/header.css [hunk 4]
```diff
@@ -1,10 +1,44 @@
...
+}
+
+a {
+  text-decoration: none;
+}
+
+li {
+  list-style-type: none;
+}
+
```
Comment (whateverfw) [inline frontend/src/components/header/header.css:33]: this seems to repeat functionality that already exists in `global.css`, please remove and verify everything still works correctly

File: frontend/src/pages/spotnet/dashboard/Dashboard.jsx [hunk 5]
```diff
@@ -27,22 +27,38 @@ const fetchCardData = async ({ walletId }) => {
...
+          <span className="ms-2 icon-text-gap text-style" style={{
+            color: card.title === 'Borrow' ? 'var(--borrow-color)' : 'var(--collateral-color)',
+          }}>
+            {card.balance}
+          </span>
+        </div>
+      </div>
+    </div>
+  </div>
+);
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/Dashboard.jsx:59]: This component should be in a separate file

**Review by whateverfw — CHANGES_REQUESTED** [review]

**Comment by PoulavBhowmick03** [comment]
@whateverfw @djeck1432 Made the changes, PTAL

**Review by whateverfw — APPROVED** [review]
Looks good to me @PoulavBhowmick03 @djeck1432

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Comment by PoulavBhowmick03** [comment]
Thanks! @djeck1432 @whateverfw

**Pull request merged**